### PR TITLE
Added ability to publish multiple themes

### PR DESCRIPTION
**Pull request by SamGodwin2**
### What is the context of this PR?
This PR allows you to publish multiple surveys to the register with different themes. Will require the latest survey register changes, that can be found at: https://github.com/ONSdigital/eq-survey-register/pull/5

### How to review 
Test the new publish page works as expected and when you publish the appropriate number of surveys show up in the register.

### CODE REVIEW

**Comment by heofs** [comment]
[Themes for business surveys](https://trello.com/c/g6p1YcK9/619-themes-for-business-surveys)

File: eq-author-api/schema/typeDefs.js [hunk 1]
```diff
@@ -1083,7 +1083,7 @@ type Subscription {
 input PublishQuestionnaireInput {
   questionnaireId: ID!
   surveyId: String!
-  formType: String!
+  formTypes: JSON!
```
Comment (heofs) [inline eq-author-api/schema/typeDefs.js]: I think we need a migration for this

File: eq-author/src/App/publish/PublishPage.js [hunk 2]
```diff
@@ -88,61 +131,84 @@ const PublishPage = ({ match, history }) => {
...
-              <Caption>Enter survey ID</Caption>
-              <Shadow>
-                <Input
-                  id="surveyId"
-                  onChange={handleInputChange}
-                  value={inputs.surveyId}
+        <StyledPanel>
+          <Field>
+            <Label>Theme</Label>
+            <Caption>Select themes to be used for this</Caption>
```
Comment (heofs) [inline eq-author/src/App/publish/PublishPage.js]:
```suggestion
            <Caption>Select themes to be used for this survey</Caption>
```
Design is different

File: eq-author/src/App/publish/PublishPage.js [hunk 3]
```diff
@@ -88,61 +131,84 @@ const PublishPage = ({ match, history }) => {
...
-              <Label htmlFor="surveyId">Survey ID</Label>
-              <Caption>Enter survey ID</Caption>
-              <Shadow>
-                <Input
-                  id="surveyId"
-                  onChange={handleInputChange}
-                  value={inputs.surveyId}
+        <StyledPanel>
+          <Field>
+            <Label>Theme</Label>
```
Comment (heofs) [inline eq-author/src/App/publish/PublishPage.js]:
```suggestion
            <Label>Themes</Label>
```

Design is slightly different for this.

File: eq-author/src/App/publish/PublishPage.js [hunk 4]
```diff
@@ -88,61 +131,84 @@ const PublishPage = ({ match, history }) => {
...
-            <InformationPanel>
-              No further changes can be made to the questionnaire after it has
-              been submitted for approval
-            </InformationPanel>
-          </AlignedColumn>
+          {!isEmpty(formTypes) && (
+            <>
+              <Field>
+                <Label>Form type</Label>
+                <Caption>Enter form type</Caption>
```
Comment (heofs) [inline eq-author/src/App/publish/PublishPage.js]:
```suggestion
                <Caption>Enter relevant form types for selected themes</Caption>
```

File: eq-author-api/schema/resolvers/base.js [hunk 5]
```diff
@@ -549,17 +549,20 @@ const Resolvers = {
...
     triggerPublish: createMutation(async (root, { input }, ctx) => {
-      const { surveyId, formType } = input;
+      const { surveyId, formTypes } = input;
       if (
         ctx.questionnaire.publishStatus !== UNPUBLISHED &&
         ctx.questionnaire.publishStatus !== UPDATES_REQUIRED
       ) {
         throw new Error("This questionnaire is not unpublished.");
       }
+      if (!surveyId || Object.keys(formTypes).some(k => !formTypes[k])) {
```
Comment (heofs) [inline eq-author-api/schema/resolvers/base.js]:
```suggestion
      if (!surveyId || some(formTypes, isEmpty)) {
```

File: eq-author/src/App/publish/PublishPage.js [hunk 6]
```diff
@@ -82,61 +125,84 @@ const PublishPage = ({ match, history }) => {
...
+            been submitted for approval
+          </InformationPanel>
           <Button
             type="submit"
             variant="primary"
-            disabled={!(inputs.surveyId && inputs.formType)}
+            disabled={
+              !surveyId ||
+              isEmpty(formTypes) ||
+              Object.keys(formTypes).some(k => !formTypes[k])
```
Comment (heofs) [inline eq-author/src/App/publish/PublishPage.js]: Disabled here can just be `disabled={!surveyId || some(formTypes, isEmpty)}`

File: eq-author-api/schema/resolvers/base.js (hunk 5 above)
Comment (SamGodwin2) [reply eq-author-api/schema/resolvers/base.js]: Nice simplification 👍

File: eq-author/src/App/publish/PublishPage.js (hunk 6 above)
Comment (SamGodwin2) [reply eq-author/src/App/publish/PublishPage.js]: Needs the isEmpty(formTypes) just is case no formtypes but the other simplification is good.

**Pull request merged**